internal/slicer/slicer.go

```diff
@@ -330,3 +339,15 @@ func contains(l []string, s string) bool {
 	}
 	return false
 }
+
+func isListedInSlice(extractInfo *deb.ExtractInfo, slice *setup.Slice) bool {
+	if extractInfo == nil {
+		return false
+	}
+	for path := range slice.Contents {
+		if extractInfo.Path == path {
+			return true
+		}
+	}
+	return false
```

Comment:
What do you think about utilising `pathInfos` to see if a path is part of a slice `contents`?

Reply:
I think the problem with using `pathInfos` is that it captures the paths for every slice. If slice A declares `foo/bar` and slice B declares `foo/` then `pathInfos = {"foo/bar", "foo/"}`. When slice A creates the parent directory `foo` it is going to be added to the report because it is in `pathInfos`.

We could add the slice provenance to `pathInfos` but then we'll be duplicating information.

Reply:
> I think the problem with using `pathInfos` is that it captures the paths for every slice. If slice A declares `foo/bar` and slice B declares `foo/` then `pathInfos = {"foo/bar", "foo/"}`. When slice A creates the parent directory `foo` it is going to be added to the report because it is in `pathInfos`.

Ah! Never mind then. It seems good to me now.

Reply:
> I think the problem with using `pathInfos` is that it captures the paths for every slice. If slice A declares `foo/bar` and slice B declares `foo/` then `pathInfos = {"foo/bar", "foo/"}`. When slice A creates the parent directory `foo` it is going to be added to the report because it is in `pathInfos`.

Ah! Never mind then. It seems good to me now.